### ISSUE TITLE
High battery usage

### ISSUE THREAD

**Issue by thohug** [datamapio/posmo_one_testing#17]
**Setting**
Zurich, Switzerland, Android

wifi-fence: 1
do_not_track: 0
wifi_over_gps: 1
data_over_wifi_only: 1

**Issue**
I have been experiencing really high battery consumation lately, not getting through a single day without charging.
![Screenshot_20200805-181703_Device care](https://user-images.githubusercontent.com/67195528/89438054-a095a400-d748-11ea-9f9b-2e04fc480ceb.jpg)

**Comment by djr4** [comment]
Hi Thomas, thanks for reporting and testing. It's really insightful. That being said we are aware of battery problem for some time now, but it's always a tradeoff accuracy vs battery. With wifi fence, idea was that once you're in place where there is wifi (and we pick up static point (place) ) we would kill all services until wifi is lost. At that moment we would reactivate Posmo.

Experiments with geofence on the other hand turned out to be really bad. (cut off segments until something is picked up could be larger then 500m no matter what bounding you would set.

With wifi fence this looks promising but requires that you connect to stable wifi once you're idle. If you have an idea, please let me/us know. 

Thanks,
Djordje

**Comment by rogerfischer** [comment]
Hi Thomas,
For how long did you move around on this day/s?

**Comment by thohug** [comment]
Shut down after around 4hours. Around 3 hours I was at office in Wifi Fence.

Well, I see the touble of starting tracking when movement is detected. I was wondering, if the motion sensors can't detect when I start moving and activate the tracking. What I've also been thinking is using a kind of wifi-fence without having to log into the wifi (for instance, if there are constantly 3 or 4 of the same wifis around, you're probably not moving - problem there are mobile hotspots I guess...).

**Comment by rogerfischer** [comment]
You are always moving unless you put your phone down. It is the machine learning that figures out if you really move or not.

**Comment by thohug** [comment]
Yes, I understand that. But when I actually put my phone down (which I do when I'm more or less stationary), there would be no need to have the tracker running (similar to the Wifi-Fence I guess), and tracking is triggered again by motion? Or to reduce the tracking rate.

For example: When I'm at the lake all saturday long, I might lock my phone up in the locker. Probably it's detecting no movement all day long, is it in that case still trying to get a GPS signal with the usual frequency? Or on the other hand, if the phone is just laying in my bag (with GPS signal), but not in motion, is it tracking with the normal frequency?

**Comment by djr4** [comment]
> Shut down after around 4hours. Around 3 hours I was at office in Wifi Fence.
> 
> Well, I see the touble of starting tracking when movement is detected. I was wondering, if the motion sensors can't detect when I start moving and activate the tracking. What I've also been thinking is using a kind of wifi-fence without having to log into the wifi (for instance, if there are constantly 3 or 4 of the same wifis around, you're probably not moving - problem there are mobile hotspots I guess...).

Yes, actually that is the idea for wifi fence v2. Kind of keep list of wifi-s with reasonable signal strength - when this wifi drops out we start tracker. (condition for killing service would be your strongest wifi is on the list for 10min or so.)
As for mobile hotspots plan is to check location distance(t-10min, t0 min) > 1km before services is turned off. (for example)

Problem would be if you have some device in your backpack all the time. 

But this also could be solved by observing strongest wifi tuple, if one is dropped we start tracker. (I guess it's really small chance that you have 2 mobile hotspots)   

@Fangio92 do we have an acceleration (phone flat down test in wifi fence) ?

**Comment by Fangio92** [comment]
No

**Comment by rogerfischer** [comment]
We should also analyze when and where the battery drops the most. @thohug which day should we check?

**Comment by rogerfischer** [comment]
@thohug - We had the "phone lying down - no data is sent" before, but we got rid of it because of Wi-Fi Fence. So it seems Wi-Fi Fence is not always working as intended. We figure this out and let you know.
Thanks for the feedback, as always.

**Comment by thohug** [comment]
> 
> 
> We should also analyze when and where the battery drops the most. @thohug which day should we check?

The screenshot above is from 5th of august. Here's anotherone from yesterday, when I was pretty much all day long inside the wifi fence at work. Left the fence around 7pm - energy consumption just seems to marginally increase.

![grafik](https://user-images.githubusercontent.com/67195528/89624826-20ca1f80-d897-11ea-8a04-b0cb4464f6a0.png)

**Comment by djr4** [comment]
@thohug this idea of listening to wifi networks (anchoring in a way) is problem on android devices since you have only 1 attempt of scanning wifi every 30min (that is from official android docs)

see: https://developer.android.com/guide/topics/connectivity/wifi-scan (under **throttling**) (for background services) 

I'll see with @Fangio92 if there is a work around but from what it looks right now connected to wifi looks like a best bet.

**Comment by thohug** [comment]
> 
> 
> @thohug this idea of listening to wifi networks (anchoring in a way) is problem on android devices since you have only 1 attempt of scanning wifi every 30min (that is from official android docs)
> 
> see: https://developer.android.com/guide/topics/connectivity/wifi-scan (under **throttling**) (for background services)
> 
> I'll see with @Fangio92 if there is a work around but from what it looks right now connected to wifi looks like a best bet.

Maybe checking for bluetooth beacons instead of wifi is a possibility (I just stumbled across this https://developers.google.com/awareness/android-api/fence-api-overview)

**Comment by thohug** [comment]
Hi & sorry for the long leave :) as I'm moving around again more often in post-lockdown (or pre-lockdown2)-times, my phone never makes a whole day (and it is quite new...) before running out of battery. Makes me turn off the tracking when I know I'm moving around a lot or if I'm not inside wifi fences.

Friday I spent the whole day within wifi fence, battery consumption was nevertheless 25% from posmo app...

**Comment by rogerfischer** [comment]
@thohug Have you downloaded the newest version? We have added a new version which stops when the phone is still and when it is in a wifi fence.

**Comment by thohug** [comment]
Yes...Am 14.09.2020 10:24 schrieb Roger <notifications@github.com>:
@thohug Have you downloaded the newest version?

**Comment by rogerfischer** [comment]
Hi @thohug 
Also, hast Du Version 301?

An welchen Tagen hattest Du den grössten Battery Drain? Dann können wir mal schauen, wo bzw. wann dieser entsteht.

**Comment by thohug** [comment]
Ich habe Build 302 drauf aktuell … Beispielsweise Freitag war extrem; Samstag etwas besser aber trotz wifi-fence 30% Akkuverbrauch.

 

 

 

Von: Roger <notifications@github.com> 
Gesendet: Montag, 14. September 2020 13:29
An: datamapio/posmo_one_testing <posmo_one_testing@noreply.github.com>
Cc: thohug <thomas-hug@gmx.ch>; Mention <mention@noreply.github.com>
Betreff: Re: [datamapio/posmo_one_testing] High battery usage (#17)

 

Hi @thohug <https://github.com/thohug> 
Also, hast Du Version 301?

An welchen Tagen hattest Du den grössten Battery Drain. Wir schauen mal, wo dieser entsteht.

**Comment by rogerfischer** [comment]
Ok. Cool. We look into it.

**Comment by thohug** [comment]
hier noch die der Akkuverbrauch an den beiden Tagen:

![Screenshot_20200914-151708_Device care](https://user-images.githubusercontent.com/67195528/93100210-36f2a900-f6a9-11ea-9cbc-177dc07769d9.jpg)
![Screenshot_20200914-151658_Device care](https://user-images.githubusercontent.com/67195528/93100215-36f2a900-f6a9-11ea-815c-2705c7b78836.jpg)
